hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MetricsCollectionCycle.java

```diff
@@ -108,21 +118,44 @@ private void collect(String name, Object source, ProbeFunction function) {
         }
     }
 
-    private void collectDouble(Object source, String name, DoubleProbeFunction doubleFunction) {
+    private void collectDouble(Object source, String name, DoubleProbeFunction function) {
+        Set<MetricTarget> excludedTargets = getExcludedTargets(function);
         try {
-            double value = doubleFunction.get(source);
-            metricsCollector.collectDouble(name, value);
+            double value = function.get(source);
+            metricsCollector.collectDouble(name, value, excludedTargets);
         } catch (Exception ex) {
-            metricsCollector.collectException(name, ex);
+            metricsCollector.collectException(name, ex, excludedTargets);
         }
     }
 
-    private void collectLong(Object source, String name, LongProbeFunction longFunction) {
+    private void collectLong(Object source, String name, LongProbeFunction function) {
+        Set<MetricTarget> excludedTargets = getExcludedTargets(function);
         try {
-            long value = longFunction.get(source);
-            metricsCollector.collectLong(name, value);
+            long value = function.get(source);
+            metricsCollector.collectLong(name, value, excludedTargets);
         } catch (Exception ex) {
-            metricsCollector.collectException(name, ex);
+            metricsCollector.collectException(name, ex, excludedTargets);
+        }
+    }
+
+    private Set<MetricTarget> getExcludedTargets(Object object) {
+        if (object instanceof ProbeAware) {
+            Probe probe = ((ProbeAware) object).getProbe();
+            return getExcludedTargets(probe);
         }
+
+        return emptySet();
+    }
+
+    private Set<MetricTarget> getExcludedTargets(Probe probe) {
+        MetricTarget[] excludedTargets = probe.excludedTargets();
+
+        if (excludedTargets.length > 0) {
+            EnumSet<MetricTarget> metricTargets = EnumSet.noneOf(MetricTarget.class);
+            Collections.addAll(metricTargets, excludedTargets);
```

Comment:
And idea for a potential enhancement that @blazember generated during an internal discussion. This enhancement should not prevent from merging this PR.

We may be able to avoid litter generated by `EnumSet` allocated in this method by caching all possible combinations of `MetricTarget[]` in `MetricTarget` enum. The cache could look like `Map<Integer, Set<MetricTarget>>` (or it could be `Int2ObjectHashMap<Set<MetricTarget>>`). Keys could be calculated based on `MetricTarget[]` elements by a hash function, e.g. something like:
```java
int hash(MetricTarget[] excluded) {
    int hash = 0;
    for (int i = 0; i < excluded.length; i++) {
        hash += excluded[i].ordinal() * Math.pow(10, i);
    }
   return hash;
}
```